dts/arm/silabs/siwg917.dtsi

```diff
@@ -59,6 +59,80 @@
 			current-speed = <115200>;
 			status = "disabled";
 		};
+
+		egpio0: egpio@46130000 {
+			compatible = "silabs,siwx917-gpio";
+			reg = <0x46130000 0x1260>;
+			interrupts = <52 0>, <53 0>, <54 0>, <55 0>,
+				     <56 0>, <57 0>, <58 0>, <59 0>;
+			interrupt-names = "PIN0", "PIN1", "PIN2", "PIN3",
+					  "PIN4", "PIN5", "PIN6", "PIN7";
+
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			gpioa: gpio@0 {
+				compatible = "silabs,siwx917-gpio-port";
+				reg = <0>;
+				gpio-controller;
+				#gpio-cells = <2>;
+				status = "okay";
+			};
+
+			gpiob: gpio@1 {
+				compatible = "silabs,siwx917-gpio-port";
+				reg = <1>;
+				gpio-controller;
+				#gpio-cells = <2>;
+				status = "okay";
+			};
+
+			gpioc: gpio@2 {
+				compatible = "silabs,siwx917-gpio-port";
+				reg = <2>;
+				gpio-controller;
+				#gpio-cells = <2>;
+				status = "okay";
+			};
+
+			gpiod: gpio@3 {
+				compatible = "silabs,siwx917-gpio-port";
+				reg = <3>;
+				gpio-controller;
+				#gpio-cells = <2>;
+				status = "okay";
+			};
+		};
+
+		egpio1: egpio@2404c000 {
+			compatible = "silabs,siwx917-gpio";
+			reg = <0x2404C000 0x1260>;
+			interrupts = <18 0>;
+			interrupt-names = "ULP";
+			silabs,ulp;
```

Comment:
Do you think it would make sense to introduce `compatible = "silabs,siwx917-gpio-ulp"` rather than this property? (BTW, I am afraid this property will be rejected by the upstream).

Reply:
Eh. It's the same GPIO IP, but different pads with different configuration options and registers. So the idea behind the property is to have a way of knowing whether to use the HP or ULP pad configuration functions. Unfortunately, the HAL also has split a bunch of pure GPIO functions into "normal" and "ulp" variants instead of taking an instance pointer, so the value of the property is sometimes used to dispatch to the relevant HAL function (rather than inlining the register write directly in the Zephyr driver). But the point still stands that it's a "real" physical difference between the two instances of the same IP, so IMO the DT property indicating this should be fine. Maybe I could rename it to `ulp_pad` or something make it more clear.